sdk/python/feast/infra/aws.py

```diff
@@ -1,23 +1,165 @@
+import base64
 import os
 import uuid
 from datetime import datetime
 from pathlib import Path
 from tempfile import TemporaryFile
+from typing import Sequence, Union
 from urllib.parse import urlparse
 
 from colorama import Fore, Style
 
 import feast
-from feast.constants import AWS_LAMBDA_FEATURE_SERVER_IMAGE
+from feast import Entity, FeatureTable, FeatureView, __version__
+from feast.constants import (
+    AWS_LAMBDA_FEATURE_SERVER_IMAGE,
+    FEAST_USAGE,
+    SERVER_CONFIG_BASE64_ENV_NAME,
+)
 from feast.errors import S3RegistryBucketForbiddenAccess, S3RegistryBucketNotExist
 from feast.infra.passthrough_provider import PassthroughProvider
+from feast.infra.utils import aws_utils
 from feast.protos.feast.core.Registry_pb2 import Registry as RegistryProto
 from feast.registry_store import RegistryStore
 from feast.repo_config import RegistryConfig
 
+try:
+    import boto3
+except ImportError as e:
+    from feast.errors import FeastExtrasDependencyImportError
+
+    raise FeastExtrasDependencyImportError("aws", str(e))
+
 
 class AwsProvider(PassthroughProvider):
-    def _upload_docker_image(self) -> None:
+    def _get_lambda_name(self):
+        return f"feast-python-server-{__version__.replace('+', '_').replace('.', '_')}"
+
+    def update_infra(
+        self,
+        project: str,
+        tables_to_delete: Sequence[Union[FeatureTable, FeatureView]],
+        tables_to_keep: Sequence[Union[FeatureTable, FeatureView]],
+        entities_to_delete: Sequence[Entity],
+        entities_to_keep: Sequence[Entity],
+        partial: bool,
+    ):
+        self.online_store.update(
+            config=self.repo_config,
+            tables_to_delete=tables_to_delete,
+            tables_to_keep=tables_to_keep,
+            entities_to_keep=entities_to_keep,
+            entities_to_delete=entities_to_delete,
+            partial=partial,
+        )
+
+        if self.repo_config.feature_server and self.repo_config.feature_server.enabled:
+            image_uri = self._upload_docker_image()
+            print("Deploying feature server...")
+
+            assert self.repo_config.repo_path
+            with open(self.repo_config.repo_path / "feature_store.yaml", "rb") as f:
+                config_bytes = f.read()
+                config_base64 = base64.b64encode(config_bytes).decode()
+
+            resource_name = self._get_lambda_name()
+            lambda_client = boto3.client("lambda")
+            api_gateway_client = boto3.client("apigatewayv2")
+            function = aws_utils.get_lambda_function(lambda_client, resource_name)
+
+            if function is None:
+                # If the Lambda function does not exist, create it.
+                print("  Creating AWS Lambda...")
+                lambda_client.create_function(
```

Comment:
AWS resources can be tagged: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/lambda.html#Lambda.Client.create_function

we should add some common tags for ease of use. I recommend something like:
```
{
    "feast-owned": 'true",
    "project": project_id
}
```

Maybe also the python version for feast. Probably also true for the API gateway or any other resource we create and expect to manage..